hnn_core/network.py

```diff
@@ -399,6 +417,51 @@ def __repr__(self):
                  len(self.pos_dict['L5_basket'])))
         return '<%s | %s>' % (class_name, s)
 
+    def _reset_cell_positions(self):
+        pos = _create_cell_coords(n_pyr_x=self._N_pyr_x, n_pyr_y=self._N_pyr_y,
+                                  zdiff=self._layer_separation,
+                                  inplane_distance=self._inplane_distance)
+        for key in pos.keys():
+            self.pos_dict[key] = pos[key]
+
+    @property
+    def inplane_distance(self):
+        return self._inplane_distance
+
+    @inplane_distance.setter
+    def inplane_distance(self, new_dist):
+        if len(self.external_drives) > 0:
+            raise RuntimeError('In-plane distance between network cells can '
+                               'only be adjusted before external drives are '
+                               f'attached; have {len(self.external_drives)} '
+                               'drives.')
+        _validate_type(new_dist, (float, int), 'new distance')
+        if not new_dist > 0.:
+            raise ValueError('In-plane distance must be positive definite, '
+                             f'got: {new_dist}')
+
+        self._inplane_distance = new_dist
+        self._reset_cell_positions()
+
+    @property
+    def layer_separation(self):
+        return self._layer_separation
+
+    @layer_separation.setter
+    def layer_separation(self, new_sep):
+        if len(self.external_drives) > 0:
+            raise RuntimeError('Layer separation between pyramidal cells can '
+                               'only be adjusted before external drives are '
+                               f'attached; have {len(self.external_drives)} '
+                               'drives.')
+        _validate_type(new_sep, (float, int), 'new separation')
+        if not new_sep > 0.:
+            raise ValueError('Layer separation must be positive definite, '
+                             f'got: {new_sep}')
```

Comment:
I would just give this as an argument to `set_cell_position` and set it inside there